functional/unit_action_test.go

```diff
@@ -404,3 +407,63 @@ func TestUnitStatus(t *testing.T) {
 			stdout, stderr)
 	}
 }
+
+// TestListUnitFilesOrder simply checks if "fleetctl list-unit-files" returns
+// an ordered list of units
+func TestListUnitFilesOrder(t *testing.T) {
+	cluster, err := platform.NewNspawnCluster("smoke")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer cluster.Destroy()
+
+	m, err := cluster.CreateMember()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	_, err = cluster.WaitForNMachines(m, 1)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Combine units
+	var units []string
+	for i := 1; i <= 20; i++ {
+		unit := fmt.Sprintf("fixtures/units/hello@%d.service", i)
+		stdout, stderr, err := cluster.Fleetctl(m, "submit", unit)
+		if err != nil {
+			t.Fatalf("Failed to submit a batch of units: \nstdout: %s\nstder: %s\nerr: %v", stdout, stderr, err)
+		}
+		units = append(units, unit)
+	}
+
+	// make sure that all unit files will show up
+	_, err = cluster.WaitForNUnitFiles(m, 20)
+	if err != nil {
+		t.Fatal("Failed to run list-unit-files: %v", err)
+	}
+
+	stdout, _, err := cluster.Fleetctl(m, "list-unit-files", "--no-legend", "--fields", "unit")
```

Comment:
Actually I had thought that it would be redundant to call ``"fleetctl list-unit-files"`` again. But after looking into the cluster operation deeper, I can confirm that actually ``cluster.WaitForNUnitFiles()`` does not return a sorted map. (Hmm, not exactly what I expected - scratching my head.) Of course we could make it return a sorted map, if we would really want it. But for now I think it's fine to simply call ``"fleetctl list-unit-files"`` again like this, as this functional test is not in a critical hot-path.
So LGTM.